File: Java_ChildrenCompetition/src/main/java/service/ITestParticipantRelationService.java
package service;

import model.HasId;

//public interface ITestParticipantRelationService <ID, T extends HasId<ID>>{
//    void save(int idTest, int idParticipant);
//}

public interface ITestParticipantRelationService {
    void save(int idTest, int idParticipant);
}

File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/service/ITestParticipantRelationService.cs
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.service
{
    public interface ITestParticipantRelationService<ID, T> where T : HasId<ID>
    {
        void save(int idTest, int idParticipant);
    }
}